### ISSUE TITLE
Expand regex option documentation across rules

### ISSUE THREAD

**Issue by rosven** [stylelint/stylelint#1650]
**BUG: selector-class-pattern** 
Does not seem to handle regex expressions properly.

When testing the example as in the documentation: 

```json
"selector-class-pattern": "/foo-[a-z]+/",
````

Both selectors fail, but only the first should fail in theory. 
```css
.foo-BAR {
  color: #FFF;
}

.foo-bar {
  color: #FFF;
}
```

It was wanted to ban following patterns in my css files since the selectors are imported as variables in javascript. 

```css
.test-class {
  color: #FFF;
}
```

And instead allow

```css
.testClass {
  color: #FFF;
}
```

stylelint version: **7.0.1**
stylelint-config-standard version: **11.0.0**

**.stylelintrc:**
```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "selector-class-pattern": "/[a-z][a-z0-9A-Z]+/",
  }
}
```

The regex `\([a-z][a-z0-9A-Z]+)\` works fine and can be tested at: http://regexr.com/

```css
.testClass {
  color: #00F;
}

.test-class {
	color: #00F;
}
```
**Expected:**
```shell
 x:x  ✖  Expected class selector ".test-class" to match specified pattern   selector-class-pattern
```

**Got:**
```shell
 x:x  ✖  Expected class selector ".testClass" to match specified pattern    selector-class-pattern
 x:x  ✖  Expected class selector ".test-class" to match specified pattern   selector-class-pattern
```

**Comment by evilebottnawi** [comment]
@rosven invalid `/foo-[a-z]+/` allow only `foo-bar`, but not `foo-BAR`, use `/foo-[a-zA-Z]+/` or create new issue with enchantment allow use `i` flag in regexp.

**Comment by evilebottnawi** [comment]
@rosven also if use string `"selector-class-pattern": "/[a-z][a-z0-9A-Z]+/"`, you can't put `/` and `/` inside string.
In rule
```javascript
const normalizedPattern = (_.isString(pattern))
      ? new RegExp(pattern)
      : pattern
```
If you use pattern `/foo-[a-z]+/` as string, you got `normalizedPattern` as `/\/foor-[a-z]+\//` and it is not valid.

**Comment by rosven** [comment]
@evilebottnawi That is correct! I was using it wrong ˆˆ, now it works. Thank you!

**Comment by evilebottnawi** [comment]
@rosven :+1: @jeddy3 do not forget about the documentation this :smile:

**Comment by jeddy3** [comment]
> @jeddy3 do not forget about the documentation this 😄

If you think we can better communicate the regex option, then we should be open this issue and label it up as "documentation". 

I'm struggling to grok exactly what's going on in this issue though, and so can explain it once more please.

**Comment by ntwb** [comment]
> I'm struggling to grok exactly what's going on in this issue though, and so can explain it once more please.

https://github.com/stylelint/stylelint/tree/master/src/rules/selector-class-pattern#options

The example given in the docs is: `E.g. /foo-[a-z]+/` <- Is that a string or a regular expression?

> Via https://github.com/stylelint/stylelint/issues/1650#issuecomment-232647979
> If you use pattern `/foo-[a-z]+/` as string, you get `normalizedPattern` as `/\/for-[a-z]+\//` and it is not valid.

My take is this is where the confusion is at, don't use `/`'s if your wanting to use a string otherwise after converting this to a regular expression as part of `normalizedPattern` you will end up with the regular express `/\/for-[a-z]+\//` which is not the same as the original string you _thought_ this would now be checking against.

So improving the docs in regard to what format to use for a regular expression vs a string should be used, primarily highlighting the fact that if you use `/` in what you think is a string it will be interpreted as a regular expression.

**Comment by jeddy3** [comment]
@ntwb Thanks for the explanation.

Does someone what to PR a documentation fix?
